### PR TITLE
Luma to rgb transform is broadcast

### PR DESCRIPTION
**Pull request by 197g**
A bit of a walk of shame for _having had_ tests but those tests didn't make sense. Not that this was easy to decipher, having the expected coefficients for each color space appear probably shaded over the fact that the test was doing the _opposite_ of the transform where they are relevant.

The derived_luminance coefficients are in the Rgb->YCbCr matrix while for Luma->Rgb we need the first column of the YCbCr->Rgb matrix (since we assume luma to represent a `luma·(1 0 0)` vector in YCbCr). That column is always `1 1 1` as the first row of its inverse, i.e. the luminance coefficients, are constructed in such a way that they sum to `1`. So yeah, expansion is in fact alway just broadcasting in all non-constant and constant luminance chromaticity derived color spaces.

@awxkee Would you give this another pair of eyes? and I feel like we should ship it with `0.25.8` since color spaces are supposed to be flagship feature for it.

Closes: #2590

### CODE REVIEW

File: src/metadata/cicp.rs [hunk 1]
```diff
@@ -1603,7 +1585,7 @@ mod tests {
         let data = [u16::MAX];
         let color = Cicp::SRGB.into_rgb();
         let rgba = color.cast_pixels::<Luma<u16>, Rgb<u8>>(&data, &no_coefficient_fallback);
-        assert_eq!(rgba, [54, 182, 18]);
+        assert_eq!(rgba, [u8::MAX; 3]);
```
Comment (awxkee) [inline src/metadata/cicp.rs]: It’s better not to test with edge values alone, since you almost always clamping. If something goes wrong in the computation you'll still just get 0 or 255, which doesn't really help.
